tests/mosaic/gpu_test.py

```diff
@@ -379,14 +379,20 @@ def kernel(ctx, inp, out, _):
       inp = memref.load(inp, [])
       mgpu.FragmentedArray.splat(inp, expected.shape, is_signed=True).store_untiled(out)
 
-    res = mgpu.as_gpu_kernel(
-        kernel,
-        (1, 1, 1),
-        (128, 1, 1),
-        jax.ShapeDtypeStruct(shape=(), dtype=expected.dtype),
-        expected,
-        (),
-    )(scalar)
+    # JAX disables 64-bit types by default. This leads to the input
+    # scalar being traced as a uint32 value. This causes the top 32
+    # bits of the 64-bit values read from the 32-bit input buffer to
+    # sometimes (nondeterministically) contain garbage.
+    # Here we explicitly reenable 64-bit values to prevent this.
+    with config.enable_x64(False):
```

Comment:
>> Here we explicitly reenable 64-bit values to prevent this.

Should this be enable_x64(True)?

Reply:
Yeah my bad, I was testing that this has an effect and copied the wrong diff :)
I pushed an updated commit.